Model/Cron.php

```diff
@@ -1232,7 +1280,61 @@ protected function _processNotification()
                     $this->_adyenLogger->addAdyenNotificationCronjob($message);
                     $comment = $this->_order->addStatusHistoryComment($message);
                     $this->_order->addRelatedObject($comment);
-                } else {
+                }
+                //store recurring contract for alternative payments methods
+                if ($this->configHelper->isStoreAlternativePaymentMethodEnabled()){
+                    $paymentTokenAlternativePaymentMethod = null;
+                    try {
+                        //get the payment
+                        $payment = $this->_order->getPayment();
+                        $customerId = $this->_order->getCustomerId();
+
+                        $this->_adyenLogger->addAdyenNotificationCronjob(
+                            '$paymentMethodCode ' . $this->_paymentMethod
+                        );
+                        if (!empty($this->_recurringDetailReference)) {
+                            // Check if $paymentTokenAlternativePaymentMethod exists already
+                            $paymentTokenAlternativePaymentMethod = $this->paymentTokenManagement->getByGatewayToken(
+                                $this->_recurringDetailReference,
+                                $payment->getMethodInstance()->getCode(),
+                                $payment->getOrder()->getCustomerId()
+                            );
+
+                            // In case the payment token for this payment method does not exist, create it based on the additionalData
+                            if ($paymentTokenAlternativePaymentMethod === null) {
+
+
+                                /** @var PaymentTokenInterface $paymentTokenAlternativePaymentMethod */
+                                $paymentTokenAlternativePaymentMethod = $this->paymentTokenFactory->create(
+                                    PaymentTokenFactoryInterface::TOKEN_TYPE_ACCOUNT
```

Comment:
Check how this changes the flow

Reply:
Switched back to credit card as Magento does not show the PM in the customer account without it